compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp

```diff
@@ -29,28 +31,50 @@ namespace {
 class PolynomialApproximationPass final
     : public impl::PolynomialApproximationPassBase<
           PolynomialApproximationPass> {
+public:
+  using Base::Base;
+
   void runOnOperation() override {
+    using PatternFunction = llvm::function_ref<void(RewritePatternSet &)>;
+    llvm::StringMap<PatternFunction> patternMap = {
+        {"tan", populateExpandTanPattern},
+        {"sinh", populateExpandSinhPattern},
+        {"cosh", populateExpandCoshPattern},
+        {"asinh", populateExpandAsinhPattern},
+        {"acosh", populateExpandAcoshPattern},
+        {"atanh", populateExpandAtanhPattern},
+        {"powf", populateExpandPowFPattern},
+        {"fpowi", populateExpandFPowIPattern},
+        {"erf",
+         [&](RewritePatternSet &mathPatterns) {
+           this->populateErfPattern(mathPatterns);
+         }},
+    };
+
     RewritePatternSet mathPatterns(&getContext());
-    populateExpandTanPattern(mathPatterns);
-    populateExpandSinhPattern(mathPatterns);
-    populateExpandCoshPattern(mathPatterns);
-    populateExpandAsinhPattern(mathPatterns);
-    populateExpandAcoshPattern(mathPatterns);
-    populateExpandAtanhPattern(mathPatterns);
-    populateExpandPowFPattern(mathPatterns);
-    populateExpandFPowIPattern(mathPatterns);
 
+    for (const auto &[fnName, populateFn] : patternMap) {
+      // Skip any ops in the "do not convert" list.
+      if (llvm::find(noApproxOps, fnName) != noApproxOps.end()) {
+        continue;
+      }
+      populateFn(mathPatterns);
+    }
+
+    if (failed(
+            applyPatternsGreedily(getOperation(), std::move(mathPatterns)))) {
+      return signalPassFailure();
+    }
+  }
+
+  void populateErfPattern(RewritePatternSet &mathPatterns) {
     if (clNativeMathPrecision) {
       mathPatterns.add<math::ErfPolynomialApproximation>(&getContext());
     } else {
       populateExpandExp2FPattern(mathPatterns);
       populateMathPolynomialApproximationPatterns(mathPatterns);
       populateExpandRoundEvenPattern(mathPatterns);
     }
```

Comment:
If you are going to leave this as-is, can you at least add a TODO to retire this flag and clean up the patterns for erf?

Reply:
added